### PR TITLE
Blazor Linux hosting

### LINKED ISSUE #24003: Deployment guidance of Hosted Blazor WASM under a sub-path
I have a hosted blazor wasm app, and I want to deploy it  under a sub-path like `https://xxx.mydomain.com/blazor` , with nginx.
I've deployed the blazor app on the server(`127.0.0.1:5001`) and finished the nginx configurations like below,
```
location /blazor {
            proxy_pass http://127.0.0.1:5001;
            limit_req  zone=one burst=60 nodelay;
        }
```
A hosted blazor wasm has two parts, server side(web api) and client side(blazor wasm). On the server side, I've configured `UsePathBase("/blazor")` in `Startup.cs`. If it's just a `ASP.NET Core Web API` app, it works. But the blazor app returns a lot of `404` http code of static files as the broswer still visits them with urls like `https://xxx.mydomain.com/assets/xxx` instead of `https://xxx.mydomain.com/blazor/assets/xxx`.

I've read issues like #14999 or #15464, but still not find a perfect answer. I think a related doc is required. 
Any help? Thanks in advance!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: be59faf1-5a20-b7e9-19ff-06589023bb15
* Version Independent ID: 250e7458-3c1d-0117-261d-c03ac136f16a
* Content: [Host and deploy ASP.NET Core Blazor WebAssembly](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#nginx)
* Content Source: [aspnetcore/blazor/host-and-deploy/webassembly.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/host-and-deploy/webassembly.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #24003

### CODE REVIEW

**Comment by guardrex** [comment]
@LeaFrock ... I'm still working on this, but you can probably comment on it at this point. I might spin up an Azure VM (or two, Nginx and Apache) to test this guidance, but it's also possible that the product unit will review and sign-off on the PR next week without requiring me to run tests. If you get a chance to look this over, I thank you in advance for any feedback.

**Comment by LeaFrock** [comment]
> I'm also concerned about the host... but you provided 0.0.0.0 ... 

Never mind, it's a typo. If nginx is deployed on the same machine as blazor app, it should be `localhost` certainly. But in my production environment,  nginx is deployed on another machine while the blazor app is hosted behind it in LAN. So it cannot use `127.0.0.1`.

**Comment by guardrex** [comment]
Thanks @LeaFrock ... I'm still working on the PR, and I hope to perform some Apache testing for the Apache bits tomorrow (Monday). Let me know if you have any further feedback on the PR.

File: aspnetcore/blazor/host-and-deploy/webassembly.md [hunk 1]
```diff
@@ -166,6 +166,222 @@ For more information, see the following articles:
...
+  <base href="{PATH}">
+  ```
+
+  For an app that loads requests at `/blazor`, the `href` attribute is set to a value of `/blazor/`. Note that the trailing slash in the following example is required:
+
+  ```html
+  <base href="/blazor/">
+  ```
+
+* In links throughout the app, do ***not*** prefix with a forward slash. Either avoid the use of a path segment separator or use dot-slash (`./`) relative path notation:
```
Comment (LeaFrock) [inline aspnetcore/blazor/host-and-deploy/webassembly.md:246]: I think `NavigationManager` should be mentioned here too. It's quite common in the development of a blazor app, though it's essentially similar to `href`.

``` html
@page "/account"
@inject NavigationManager NavigationManager

<button onclick='Navi2OtherPage'>JumpToOtherPage</button>

@code{
    private void Navi2OtherPage()
    {
        NavigationManager.NavigateTo("/other"); //Incorrect
        NavigationManager.NavigateTo("other"); //Correct
        NavigationManager.NavigateTo("./other"); //Correct
    }
}
```

```

**Comment by guardrex** [comment]
@LeaFrock ... I'm still working on this a little. Management didn't make a decision on Apache coverage yet, so I'm going to comment-out the Apache part for later. I'll work on this topic again for an upcoming overhaul on https://github.com/dotnet/AspNetCore.Docs/issues/19286. I'll work on the Apache piece later. For now, I might make a few more small changes, then I'll get this merged. Thanks again for your help and feedback on the PR.

**Comment by guardrex** [comment]
Ok ... I think we're good here for now. I'll be back to work on these a bit more on UE ("user experience") updates later, and we'll continue to receive feedback from the readers on them. Thanks again @LeaFrock! 🎸

**Pull request merged**